docs/monitors/appmesh.md

```diff
@@ -0,0 +1,145 @@
+<!--- GENERATED BY gomplate from scripts/docs/monitor-page.md.tmpl --->
+
+# appmesh
+
+This monitor starts a StatsD monitor to listen to StatsD metrics emitted
+by AWS AppMesh Envoy Proxy.
+
+To report AppMesh Envoy metrics, you need to enable Envoy StatsD sink on AppMesh
+and deploy the agent as a sidecar in the services that need to be monitored.
+
+
+Sample Envoy StatsD configuration:
+
+```yaml
+stats_sinks:
+ -
+  name: "envoy.statsd"
+  config:
+   address:
+    socket_address:
+     address: "127.0.0.1"
+     port_value: 8125
+     protocol: "UDP"
+   prefix: statsd.appmesh
+```
+Please remember to provide the prefix to the agent monitor configuration.
+
+See [Envoy API reference](https://www.envoyproxy.io/docs/envoy/latest/api-v2/config/metrics/v2/stats.proto#envoy-api-msg-config-metrics-v2-statsdsink) for more info
+
+Sample SignalFx SmartAgent configuration:
+
+```yaml
+monitors:
+ - type: appmesh
+   listenAddress: 0.0.0.0
+   listenPort: 8125
+   metricPrefix: statsd.appmesh
+```
+
+
+Monitor Type: `appmesh`
+
+[Monitor Source Code](https://github.com/signalfx/signalfx-agent/tree/master/internal/monitors/appmesh)
+
+**Accepts Endpoints**: No
+
+**Multiple Instances Allowed**: Yes
+
+## Configuration
+
+| Config option | Required | Type | Description |
+| --- | --- | --- | --- |
+| `listenAddress` | no | `string` | The host/address on which to bind the UDP listener that accepts statsd datagrams (**default:** `localhost`) |
+| `listenPort` | no | `integer` | The port on which to listen for statsd messages (**default:** `8125`) |
+| `metricPrefix` | no | `string` | A prefix in metric names that needs to be removed before metric name conversion |
+
+
+
+
+## Metrics
+
+The following table lists the metrics available for this monitor. Metrics that are marked as Included are standard metrics and are monitored by default.
+
+| Name | Type | Included | Description |
+| ---  | ---  | ---    | ---         |
+| `gauge.<ingress/egress>.circuit_breakers.<priority>.cx_open` | gauge | ✔ | Whether the connection circuit breaker is closed (0) or open (1) |
```

Comment:
Would be much better if metric names didn't have "gauge"/"derive" in front of them.

Reply:
Also, can ingress/egress just be a dimenison?

Reply:
Also, can ingress/egress just be a dimenison? -> Sure

Reply:
"gauge"/"derive" -> this is to match collectd/statsd plugin.

Reply:
> Also, can ingress/egress just be a dimenison? -> Sure

If it is a dimension, wouldn't the metric names change now?

Reply:
Oh, docs are out of date.